internal/network/manifests_generator.go

```diff
@@ -238,6 +239,126 @@ func (m *ManifestsGenerator) AddSchedulableMastersManifest(ctx context.Context,
 	return nil
 }
 
+const (
+	diskEncryptionManifest = `apiVersion: machineconfiguration.openshift.io/v1
+kind: MachineConfig
+metadata:
+  name: {{.NAME}}
+  labels:
+    machineconfiguration.openshift.io/role: {{.ROLE}}
+spec:
+  config:
+    ignition:
+      version: 3.2.0
+    storage:
+      luks:
+        - name: root
+          device: /dev/disk/by-partlabel/root
+          clevis:
+		  {{ if .ENCRYPTION_MODE_TPMV2 }}
+            tpm2: true
+		  {{ end }}
+		  {{ if .ENCRYPTION_MODE_TANG }}
+			tang:
+			  - url: http://tang.example.com:7500
+			    thumbprint: PLjNyRdGw03zlRoGjQYMahSZGu9
+		  {{ end }}
+          options: [--cipher, aes-cbc-essiv:sha256]
+          wipeVolume: true
+      filesystems:
+        - device: /dev/mapper/root
+          format: xfs
+          wipeFilesystem: true
+          label: root
+{{ if .ENCRYPTION_MODE_TANG }}
+  kernelArguments:
+    - rd.neednet=1
+{{ end }}
+`
+)
+
+func (m *ManifestsGenerator) AddDiskEncryptionManifest(ctx context.Context, log logrus.FieldLogger, c *common.Cluster) error {
+
+	if *c.DiskEncryption == (models.DiskEncryption{}) {
+		return nil
+	}
+
+	createDiskEncryptionManifest := func(manifestParams map[string]string) error {
+
+		log.Infof("Creating manifest to encrypt installation disk on %s nodes", manifestParams["ROLE"])
+
+		content, err := fillTemplate(manifestParams, diskEncryptionManifest, log)
+		if err != nil {
+			log.WithError(err).Errorf("Failed to parse disk encryption manifest's template for %s nodes", manifestParams["ROLE"])
+			return err
+		}
+
+		if err := m.createManifests(ctx, c, "encrypt-disk.yaml", content); err != nil {
+
+			log.WithError(err).Errorf("Failed to create manifest to encrypt disk on %s nodes", manifestParams["ROLE"])
+			return err
+		}
+
+		return nil
+	}
+
+	manifestParams := map[string]string{}
+
+	switch c.DiskEncryption.Mode {
+
+	case models.DiskEncryptionModeTpmv2:
+
+		manifestParams["ENCRYPTION_MODE_TPMV2"] = "nonEmpty"
+
+	case models.DiskEncryptionModeTang:
+
+		manifestParams["ENCRYPTION_MODE_TANG"] = "nonEmpty"
+
+	default:
+
+		return errors.New("Non of the encryption modes were specified")
+	}
+
+	switch c.DiskEncryption.EnableOn {
+
+	case models.DiskEncryptionEnableOnAll:
+
+		manifestParams["ROLE"] = "master"
+		if err := createDiskEncryptionManifest(manifestParams); err != nil {
+			return err
+		}
+
+		manifestParams["ROLE"] = "worker"
+		if err := createDiskEncryptionManifest(manifestParams); err != nil {
```

Comment:
Sp all manifest params are the same, just different ROLE.
Seems like this will override the manifest created in line 327.

Reply:
`Sp` ?

I will fix it, thanks.

Reply:
Fixed.